WordPress/Classes/ViewRelated/Reader/ReaderFollowedSitesViewController.swift

```diff
@@ -0,0 +1,334 @@
+    private var tableViewHandler: WPTableViewHandler!
+    private var tableViewController: UITableViewController!
+
+    private let cellIdentifier = "CellIdentifier"
+
+    lazy var noResultsView: WPNoResultsView = {
+        let title = NSLocalizedString("No Sites", comment: "Title of a message explaining that the user is not currently following any blogs in their reader.")
+        let message = NSLocalizedString("You are not following any sites yet. Why not follow one now?", comment: "A suggestion to the user that they try following a site in their reader.")
+        return WPNoResultsView(title: title, message: message, accessoryView: nil, buttonTitle: nil)
+    }()
+
+
+    /// Convenience method for instantiating an instance of ReaderFollowedSitesViewController
+    ///
+    /// - Returns: An instance of the controller
+    ///
+    class func controller() -> ReaderFollowedSitesViewController {
+        let storyboard = UIStoryboard(name: "Reader", bundle: NSBundle.mainBundle())
+        let controller = storyboard.instantiateViewControllerWithIdentifier("ReaderFollowedSitesViewController") as! ReaderFollowedSitesViewController
+        return controller
+    }
+
+
+    // MARK: - LifeCycle Methods
+
+
+    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
+        tableViewController = segue.destinationViewController as? UITableViewController
+    }
+
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.title = NSLocalizedString("Followed Sites", comment: "Page title for the list of followed sites.")
+        setupTableView()
+        setupTableViewHandler()
+        configureSearchBar()
+
+        WPStyleGuide.configureColorsForView(view, andTableView: tableView)
+    }
+
+
+    override func viewWillAppear(animated: Bool) {
+        super.viewWillAppear(animated)
+
+        configureNoResultsView()
+        syncSites()
+    }
+
+
+    // MARK: - Setup
+
+
+    private func setupTableView() {
+        assert(tableViewController != nil, "The tableViewController must be assigned before configuring the tableView")
+
+        tableView = tableViewController.tableView
+        WPStyleGuide.resetReadableMarginsForTableView(tableView)
+
+        refreshControl = tableViewController.refreshControl!
+        refreshControl.addTarget(self, action: #selector(ReaderStreamViewController.handleRefresh(_:)), forControlEvents: .ValueChanged)
+    }
+
+
+    private func setupTableViewHandler() {
+        assert(tableView != nil, "A tableView must be assigned before configuring a handler")
+
+        tableViewHandler = WPTableViewHandler(tableView: tableView)
+        tableViewHandler.delegate = self
+    }
+
+
+    // MARK: - Configuration
+
+
+    func configureSearchBar() {
+        let placeholderText = NSLocalizedString("Enter the URL of a site to follow", comment: "Placeholder text prompting the user to type the name of the URL or tag they would like to follow.")
```

Comment:
Sorry, this comment probably needs updating too (to remove the 'tag' reference)!